test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Test
+    public void addCommand_invalidEmail_throws() {
+        final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@",
+                "@invalid@email", "invalid@email!", "!invalid@email" };
+        for (String email : invalidEmails) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, email, false,
+                    Address.EXAMPLE, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidAddress_throws() {
+        final String[] invalidAddresses = { "", " " };
+        for (String address : invalidAddresses) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    address, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidTags_throws() {
+        final String[][] invalidTags = { { "" }, { " " }, { "'" }, { "[]\\[;]" }, { "validTag", "" },
+                { "", " " } };
+        for (String[] tags : invalidTags) {
+            Set<String> tagsToAdd = new HashSet<>(Arrays.asList(tags));
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, tagsToAdd);
+        }
+    }
+
+    /**
+     * Asserts that attempting to construct an add command with the supplied
+     * invalid data throws an IllegalValueException
+     */
+    private void assertConstructingInvalidAddCmdThrows(String name, String phone, boolean isPhonePrivate,
+            String email, boolean isEmailPrivate, String address, boolean isAddressPrivate,
+            Set<String> tags) {
+        try {
+            new AddCommand(name, phone, isPhonePrivate, email, isEmailPrivate, address, isAddressPrivate,
+                    tags);
+        } catch (IllegalValueException e) {
+            return;
+        }
+        String error = String.format(
+                "An add command was successfully constructed with invalid input: %s %s %s %s %s %s %s %s",
+                name, phone, isPhonePrivate, email, isEmailPrivate, address, isAddressPrivate, tags);
+        fail(error);
+    }
+
+    @Test
+    public void addCommand_validData_correctlyConstructed() throws Exception {
+        AddCommand cmd = new AddCommand(Name.EXAMPLE, Phone.EXAMPLE, true, Email.EXAMPLE, false,
+                Address.EXAMPLE, true, NO_TAGS); // should never throw
+        ReadOnlyPerson p = cmd.getPerson();
+
+        assertEquals(Name.EXAMPLE, p.getName().fullName);
+        assertEquals(Phone.EXAMPLE, p.getPhone().value);
+        assertTrue(p.getPhone().isPrivate());
+        assertEquals(Email.EXAMPLE, p.getEmail().value);
+        assertFalse(p.getEmail().isPrivate());
+        assertEquals(Address.EXAMPLE, p.getAddress().value);
+        assertTrue(p.getAddress().isPrivate());
+        boolean isTagListEmpty = !p.getTags().iterator().hasNext();
+        assertTrue(isTagListEmpty);
```

Comment:
That could be simply a 'shortcut' taken by the developer to save time (it was done in a rush :-).
Anyway, I think Person needs multiple equality methods each with a different level of strictness.
Sometimes we want to check if both objects contain the same exact values (I think that's the one you want), and other times we just want to check if it 'is the same person' (i.e. just compare the identity-related fields such as the name).  You can do this bit as a separate PR. Just remember to add an issue (and a `TODO` in code) so that we don't forget about it.